docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,403 @@
+
+The key provider service program encrypts the `PrivateLayerBlockCipherOptions` with the user's OpenPGP key. After that, it will perform Base64 encoding and then package it into a JSON structure. Under the confidential container system, it is called 'annotation packet' to record some additional information, such as the ID of the user's private key, for example:
+
+```
+{
+    "key_id":"1234",
+    "wrapped_data":#base64encode(Enc(PrivateLayerBlockCipherOptions)),
+    "wrap_type":"",
+}
+```
+
+The specific format of the annotation packet can be determined by the key provider program used by the user, but it is necessary to ensure that the key provider program used by the decryptor (i.e. the attestation-agent integrated with the specified Key Broker Client) supports parsing annotation packets of the same format.
+
+After completing the above work, Base64 encode the `PublicLayerBlockCipherOptions` and annotation packet, and write them into the manifest.json of the container image as the annotation field of the layer, the former is placed in the `org.opencontainers.image.enc.pubopts` field and the latter in the `org.opencontainers.image.enc.keys.provider.attestation-agent` field. Finally, add the "+encrypted" suffix to the mediaType field of this layer. For example, the container image manifest with an encrypted layer is as follows:
+
+```json
+{
+    "schemaVersion": 2,
+    "config": {
+        "mediaType": "application/vnd.oci.image.config.v1+json",
+        "digest": "sha256:ef87d9f0...",
+        "size": 1510
+    },
+    "layers": [
+        {
+            "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip+encrypted",
+            "digest": "sha256:f69ae40...",
+            "size": 204812,
+            "annotations": {
+                "org.opencontainers.image.enc.pubopts": "eyJjaXBoZXIiOi...",
+                "org.opencontainers.image.enc.keys.provider.attestation-agent": "eyJwcm9..."
+            }
+        }
+    ]
+}
+```
+
+### Image signing
+
+The container image signing standard used by the confidential containers V1 is follow the signing mechanism used in [containers/image](https://github.com/containers/image). At present, the only defined cryptographic signature format is a OpenPGP signature (RFC 4880). The signing process will not modify any content of the container image, including layer, config, manifest, etc. Therefore, the signature will not change the image manifest digest used to identify the image.
```

Comment:
I'm a bit confused by this paragraph. On line 7, you wrote "It should be noted that the encryption process must be carried out before the signing process". Should I deduce that the signing is performed on the _encrypted_ data? But then, that seems to conflict with "the signature will not change the image manifest digest"… I suspect that @jodh-intel has a similar question when he asks below "_Is 817a12c32a39be.. the sha256 value for the value of busybox:latest at the time this file was created?_", which I interpret as "is it the sha256 for the unencrypted image" (sorry if I misunderstood the question).

Reply:
When deploying an image, the signature verification action occurs before the decryption action of the image layer. Therefore, when creating an image, the signature should be performed after encryption. Therefore, in fact, the manifest and reference as the signature object are for the encrypted container image.

Reply:
@jialez0 Thanks for the clarification. Since it looks like both @jodh-intel and myself were equally confused by the wording above, let me suggest to add exactly the explanation you just gave after the text above?